packages/@lwc/engine-core/src/framework/decorators/api.ts

```diff
@@ -74,15 +74,10 @@ export function createPublicAccessorDescriptor(
     descriptor: PropertyDescriptor
 ): PropertyDescriptor {
     const { get, set, enumerable, configurable } = descriptor;
-    if (!isFunction(get)) {
-        if (process.env.NODE_ENV !== 'production') {
-            assert.invariant(
-                isFunction(get),
-                `Invalid compiler output for public accessor ${toString(key)} decorated with @api`
-            );
-        }
-        throw new Error();
-    }
+    assert.invariant(
+        isFunction(get),
+        `Invalid compiler output for public accessor ${toString(key)} decorated with @api`
+    );
```

Comment:
If you look at the logic above, this is doing the same thing. It was throwing in both dev mode and prod mode.